flow/util/misc.py

```diff
@@ -303,3 +306,46 @@ def __iter__(self):
 
     def __len__(self):
         return len(self._data)
+
+
+def _get_parallel_executor(parallelization="thread"):
+    """Get an executor for the desired parallelization strategy.
+
+    Parameters
+    ----------
+    parallelization : str
+        Parallelization mode. Allowed values are "thread", "process", or
+        "none". (Default value = "thread")
+
+    """
+    if parallelization == "thread":
+        parallel_executor = thread_map
+    elif parallelization == "process":
```

Comment:
I'd like help on this if we decide to keep process parallelism. This code just hangs when the parallel processes fork and I can't figure out why.

Reply:
I believe I've resolved this error, although at the moment a lot of tests are failing due to the recent changes in mocking. I had to check out [fa6dad6](https://github.com/glotzerlab/signac-flow/pull/417/commits/fa6dad6a94e5aa899cf8677b7d22d4cfe08cfe60) to get enough status tests to run to identify the issue, so you may need to recover to at least that state before you can validate that my fix resolves your problem.

The problem you're running into is that while you were pickling all the _arguments_ to the function, the function itself that you're trying to execute in parallel is not pickleable. This is due in part to the fact that it's a local function, but also to the fact that even if you remove it from the local scope it will be calling the function passed to `_get_parallel_executor`, which is typically going to call other unpickleable things like class methods of `FlowProject`. I resolved this issue by using cloudpickle to serialize the function itself, then binding that to a partial that can be executed.

Reply:
@vyasr Just wanted to offer a massive THANKS for this! I was having a lot of difficulty identifying that problem. Your explanation totally makes sense. Here's a cookie to express my appreciation. 🍪

Reply:
Coooookieeeee